python/GafferSceneTest/SceneReaderTest.py

```diff
@@ -492,5 +493,35 @@ def testComputeSetInEmptyScene( self ) :
 		r1 = GafferScene.SceneReader()
 		self.assertEqual( r1["out"].set( "blahblah" ).value.paths(), [] )
 
+	def testScenePathNotAvailableInContextExpressions( self ) :
+
+		s = IECore.SceneCache( self.__testFile, IECore.IndexedIO.OpenMode.Write )
+
+		sphereGroup = s.createChild( "sphereGroup" )
+		sphere = sphereGroup.createChild( "sphere" )
+		sphere.writeObject( IECore.SpherePrimitive(), 0 )
+
+		planeGroup = s.createChild( "planeGroup" )
+		plane = planeGroup.createChild( "plane" )
+		plane.writeObject( IECore.MeshPrimitive.createPlane( IECore.Box2f( IECore.V2f( -1 ), IECore.V2f( 1 ) ) ), 0 )
+
+		del s, sphereGroup, sphere, planeGroup, plane
+
+		script = Gaffer.ScriptNode()
+
+		script["reader"] = GafferScene.SceneReader()
+
+		script["expression"] = Gaffer.Expression()
+		script["expression"].setExpression( inspect.cleandoc(
+			"""
+			assert( context.get( "scene:path", None ) is None )
+			parent["reader"]["fileName"] = "{testFilePath}"
+			""".format(testFilePath=self.__testFile)
+		) )
+
+		self.assertEqual( set( [ str( x ) for x in script["reader"]["out"].childNames( "/" ) ] ), set( [ "sphereGroup", "planeGroup" ] ) )
+		self.assertEqual( set( [ str( x ) for x in script["reader"]["out"].childNames( "/sphereGroup" ) ] ), set( [ "sphere" ] ) )
+		self.assertEqual( set( [ str( x ) for x in script["reader"]["out"].childNames( "/planeGroup" ) ] ), set( [ "plane" ] ) )
+
```

Comment:
Could we add a `self.assertSceneValid( script["reader"] )` check here? That tests a whole bunch of stuff an will probably have more stuff added in the future, so is a good way of exercising as much as possible.